### PR TITLE
feature/before after hooks separation

### PR DESCRIPTION
**Pull request by ramiloif**
Hi ,
I added separation between before and after hooks ,
The reason is I think i the report the before hooks should be shown before the tests in the suite 
and the after hooks should be after.
This PR comes with PR on the report generator that depends on this changes.
In the generator PR I'll add an example screenshot

### CODE REVIEW

**Comment by adamgruber** [comment]
I'm wondering if it makes more sense to just include all hooks and let the report generator determine what to show based on a new option. Something like:
```js
showHooks: ['always', 'never', 'failed']
```
I'm thinking this could be a selection made in the report nav menu similar to how we can turn test types on and off.

**Comment by ramiloif** [comment]
You mean to include skipped hooks ?  Yes I think that's a good idea to let user desire about them .

**Comment by ramiloif** [comment]
However about the separation between before and after hooks (each of them can be either failed or skipped) is should be in mochawesome

File: src/utils.js [hunk 1]
```diff
@@ -178,7 +182,9 @@ function getFailedHooks(suite) {
  */
 function cleanSuite(suite, totalTestsRegistered) {
   suite.uuid = uuid.v4();
-  const failedHooks = _.map(getFailedHooks(suite), cleanTest);
+  const hooks = getFailedHooks(suite);
+  const beforeFailedHooks = _.map(hooks.before, cleanTest);
+  const afterFailedHooks = _.map(hooks.after, cleanTest);
```
Comment (adamgruber) [inline src/utils.js]: I'd like to see this just be
```js
const beforeHooks = _.map([].concat(suite._beforeAll, suite._beforeEach), cleanTest);
const afterHooks = _.map([].concat(suite._afterAll, suite._afterEach), cleanTest);
```
It's simpler and the renderer can handle which hooks are shown.

Comment (ramiloif) [inline src/utils.js]: Ok i'll change that
About the  - ```hasBeforeFailedHooks``` and ```hasAfterFailedHooks```
Do you think it should be changed to ```hasBeforeHooks``` and ```hasAfterHooks``` as well ?

Comment (adamgruber) [inline src/utils.js]: Yep that would make sense.

**Comment by ramiloif** [comment]
I added all hooks to the report and changed the tests as needed,
About the new config option can you guide me how to implement it ? 
More specific How to add to config a default option and how to pass the option from report to the generator ? 
I think ```showHooks -> failed``` should be the default option.

**Comment by adamgruber** [comment]
@ramiloif We're almost there with this but in the interest of saving some back and forth I'd like to update the PR to clean things up and update some of the test coverage. I'm also going to rebase my changes to clean up the commit history here. Once we get this sorted we can move on to updating the report generator. Let me know if you have any strong objections to this, otherwise can you check that you have enabled maintainer access for this PR. https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/

**Comment by ramiloif** [comment]
Allow edits from maintainers is on now

**Comment by adamgruber** [comment]
@ramiloif For some reason I wasn't able to push my commits to your branch so I just opened #158 and I'm going to close this.

**Pull request closed without merging**